docs/tutorials/client-side-encryption.txt

```diff
@@ -241,12 +241,205 @@ or creating a schema map, see later sections of this tutorial.
+        }
+      }
+    }
+  )
+
+  data_key_id = client_encryption.create_data_key(
+    'aws',
+    {
+      master_key: {
+        region: 'REGION-OF-YOUR-MASTER-KEY',
+        key: 'ARN-OF-YOUR-MASTER-KEY'
+      }
+
+    }
+  )
+
+See the `AWS Master Key`_ section of this tutorial  for more information about
+generating a new master key on AWS and finding the information you need to
+create data keys.
+
+For more information about creating a data key, see the MongoDB manual.
+
+.. seealso::
+
+  :manual:`Create a Data Encryption Key </ecosystem/use-cases/client-side-field-level-encryption-guide/#b-create-a-data-encryption-key>`
 
 Creating a Schema Map
 ---------------------
-Documentation coming soon.
+
+.. note::
+
+  Schema maps are only used in automatic encryption.
+
+Once you have created a data key, you can use it to encrypt and decrypt data
+during automatic encryption by referencing it in a schema map. A schema map
+is a Hash that provides the ``Mongo::Client`` with information about which
+fields to automatically encrypt and decrypt.
+
+The code snippet at the top of this tutorial demonstrates creating a schema
+map using a Ruby ``Hash``. While this will work, schema maps can grow quite
+large and it could be unweildy to include them in your Ruby code. Instead, it is
+recommended that you store them in a separate JSON (JavaScript Object Notation)
+file.
+
+Before creating the JSON file, Base64-encode the UUID of the your data key.
+
+.. code-block:: ruby
+
+  require 'base64'
+
+  # Take note of the Base64-encoded uuid
+  Base64.encode64(data_key_id)
+
+Then, create a new JSON file containing your schema map in the extended JSON
+format defined by the :manual:`Extended JSON v2 Documentation`</reference/mongodb-extended-json>`.
+
+Note that:
+
+* ``encryption_db`` and ``encryption_coll`` should be replaced with the
+  names of the database and collection where you plan to store encrypted data.
+* ``encrypted_field`` should be replaced with the name of the field you want to encrypt.
+* ``"bsonType": "string"`` should be replaced with the data type you intend to
+  encrypt, such as ``"bsonType": "integer"`` or ``"bsonType": "symbol"``.
+
+.. code-block:: json
+
+  {
+    "encryption_db.encryption_coll": {
+      "properties": {
+        "encrypted_field": {
+          "encrypt": {
+            "keyId": [{
+              "$binary": {
+                "base64": "YOUR-BASE64-ENCODED-DATA-KEY-ID",
+                "subType": "04"
+              }
+            }],
+            "bsonType": "string",
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+          }
+        }
+      },
+      "bsonType": "object"
+    }
+  }
+
+When you intend to use your schema map, convert it to a Ruby ``Hash`` using the
+``BSON::ExtJSON`` module in the ``bson`` Ruby gem.
+
+.. code-block:: ruby
+
+  require 'bson'
+
+  # schema_map is a Ruby Hash
+  schema_map = BSON::ExtJSON.parse(File.open('/path/to/your/file.json'))
```

Comment:
Does this work in both default and :bson parsing mode?

Reply:
It does! Is there one you would recommend over the other?

Reply:
Let's leave it the way it is for now.